media/src/objects/Function.svelte

```diff
@@ -265,10 +265,32 @@
 
     const tol = 1e-12; //tolerance for comparisons
 
+    /**
+     * Returns true if given params are valid.
+     */
+    const checkValidParams = function(a, b, c, d) {
+        if (a === b) {
+            console.error('a and b cannot be equal');
+            return false;
+        }
+
+        if (c === d) {
+            console.error('c and d cannot be equal');
+            return false;
+        }
+
+        return true;
+    };
```

Comment:
I think we are also prohibiting `a` and `c` values greater than `b` and `d` respectively. So the error message might be 'a must be less than b' and the same for `c` and `d`.

Also, the console error is nice, but shouldn't we have a way to notify the users directly. Like a red message that pops up above the input if the entry is wrong.

Reply:
The graph still displays something valid when a is greater than b, so I am allowing that for now. Whether we want to restrict that is a question for Drew.

Reply:
User-facing error messaging can be addressing in our design pass with Marc, I just want to fix the actual bugs here for now.